### DISCUSSION TITLE
Helix as my only daily driver - My 3 must have features

### DISCUSSION THREAD

**Discussion by joaocarvalhoopen** [helix-editor/helix#4567] [General]
Hello,

currently I use 3 text editors in my Linux setup.

Visual Studio Code - For code editing (Rust, Python, C, C++, Javascript) and editing MarkDown files.

Sublime Text -  For small notes that I want to have always present.

Geany editor - To write my texts with spell check in English and Portuguese, with yellow/greenish letters and black background. Easy on the eyes.

Currently there are 3 features that are lacking on Helix to be my only daily driver:

**To substitute Visual Studio Code**: 
  Have the Rust types of type inference written in place in unmodifiable text. 
  Have English spell check into identifiers and comments.
                                  
**To substitute Sublime Text**:
   To have text line wrapping for text files and markdown files.
   
**To substitute Geany editor**:
   To interface with spell checkers on Linux, to have Spell check for text in English and Portuguese.    
   To have text line wrapping.

and that's all,
Other than that features, it seams a perfect match to my needs, and I like very much the key-bindings of it, I have already done the tutorial, and liked very much the idea of the tutorial.

The plugin system to be perfect would be in Rust and not in LISP, but in another life, a long time ago, I programmed a little in LISP and Scheme :-)

Best regards,
João

**Comment by CptPotato** [comment]
There are existing issues / discussions:

Inlay hints and virtual text: #411
Soft wrap: #136 
Spellchecking discussion: #3637

**Comment by David-Else** [comment]
There is a spell checker based on Hunspell being developed: https://github.com/helix-editor/helix/pull/3965 that should work with comments (I have not tried it).

I highly recommend [ltex-ls](https://github.com/valentjn/ltex-ls) as a spelling and grammar checker for Markdown, you can use it right now by adding this to `languages.toml`:
```
[[language]]
name = "markdown"
language-server = { command = "ltex-ls" }
```

**Comment by joaocarvalhoopen** [comment]
Thank you CptPotato and David-Else for the information.
Currently on Linux I use Hunspell dictionaries for English and Portuguese.

Best regards,
João

**Comment by Omnikar** [comment]
I don't know if you know about this already but while you wait for inlayed type hints, you can use `<space>k` with the cursor on a variable to see its type (and you can use it on other symbols to bring up type info and relevant documentation).

**Comment by joaocarvalhoopen** [comment]
Thank you Omnikar, but I'm aware of that.
